server/services/property.service.js

```diff
@@ -525,3 +525,96 @@ export const deleteFloorPlan = async (floorPlanDetails) => {
     throw new ErrorHandler(httpStatus.BAD_REQUEST, 'Error deleting floor plan', error);
   }
 };
+
+export const addImage = async (imageDetails) => {
+  const property = await getPropertyById(imageDetails.propertyId).catch((error) => {
+    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  });
+
+  if (!property) {
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'Invalid property');
+  }
+
+  if (imageDetails.vendorId.toString() !== property.addedBy.toString()) {
+    throw new ErrorHandler(httpStatus.FORBIDDEN, 'You are not permitted to perform this action');
+  }
+
+  let { gallery } = property;
+
+  gallery = gallery.filter((image) => image.title === imageDetails.title);
```

Comment:
The definition of gallery on line 542 can be avoided:

```suggestion
  const gallery = property.gallery.filter((image) => image.title === imageDetails.title);
```